src/app/dataset-view/additional-components/metadata-component/components/final-yaml-modal/final-yaml-modal.component.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { CUSTOM_ELEMENTS_SCHEMA } from "@angular/core";
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { NgbActiveModal } from "@ng-bootstrap/ng-bootstrap";
+import { ApolloModule } from "apollo-angular";
+import { FinalYamlModalComponent } from "./final-yaml-modal.component";
+import { AppDatasetCreateService } from "src/app/dataset-create/dataset-create.service";
+import { emitClickOnElement } from "src/app/common/base-test.helpers.spec";
+import { DatasetInfo } from "src/app/interface/navigation.interface";
+import { Observable } from "rxjs";
+
+const testDatasetInfo: DatasetInfo = {
+    accountName: "testAccountName",
+    datasetName: "testDatasetName",
+};
+
+describe("FinalYamlModalComponent", () => {
+    let component: FinalYamlModalComponent;
+    let fixture: ComponentFixture<FinalYamlModalComponent>;
+    let createDatasetService: AppDatasetCreateService;
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            declarations: [FinalYamlModalComponent],
+            schemas: [CUSTOM_ELEMENTS_SCHEMA],
+            providers: [NgbActiveModal],
+            imports: [ApolloModule],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(FinalYamlModalComponent);
+        createDatasetService = TestBed.inject(AppDatasetCreateService);
+        component = fixture.componentInstance;
+
+        component.yamlTemplate = "test yaml";
+        component.datasetInfo = testDatasetInfo;
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+    });
+
+    it("should save event", () => {
+        const commitEventToDatasetSpy = spyOn(
+            createDatasetService,
+            "commitEventToDataset",
+        ).and.returnValue(new Observable());
+        emitClickOnElement(fixture, '[data-test-id="save-event"]');
+        expect(commitEventToDatasetSpy).toHaveBeenCalledTimes(1);
+    });
```

Comment:
Not checking the modal is closed after this event

Reply:
Added check for window close